src/arch_ppc.py

```diff
@@ -0,0 +1,862 @@
+        ]
+    ]
+    all_regs = (
+        saved_regs
+        + temp_regs
+        + [stack_pointer_reg]
+        + [
+            Register(r)
+            for r in [
+                # TODO: These `crX` registers are only used to parse instructions, but
+                # the instructions that use these registers aren't implemented yet.
+                "cr0",
+                "cr1",
+                "cr2",
+                "cr3",
+                "cr4",
+                "cr5",
+                "cr6",
+                "cr7",
+            ]
+        ]
+    )
+
+    aliased_regs: Dict[str, Register] = {}
+
+    uses_delay_slots = False
+
+    @staticmethod
+    def is_branch_instruction(instr: Instruction) -> bool:
+        return (
+            instr.mnemonic
+            in [
+                "ble",
+                "blt",
+                "beq",
+                "bge",
+                "bgt",
+                "bne",
+                "bdnz",
+                "bdz",
+            ]
+            or PpcArch.is_constant_branch_instruction(instr)
+        )
+
+    @staticmethod
+    def is_branch_likely_instruction(instr: Instruction) -> bool:
+        return False
+
+    @staticmethod
+    def get_branch_target(instr: Instruction) -> JumpTarget:
+        label = instr.args[-1]
+        if isinstance(label, AsmGlobalSymbol):
+            return JumpTarget(label.symbol_name)
+        if not isinstance(label, JumpTarget):
+            raise DecompFailure(
+                f'Couldn\'t parse instruction "{instr}": invalid branch target'
+            )
+        return label
+
+    @staticmethod
+    def is_constant_branch_instruction(instr: Instruction) -> bool:
+        return instr.mnemonic == "b"
+
+    @staticmethod
+    def is_jump_instruction(instr: Instruction) -> bool:
+        # (we don't treat jal/jalr as jumps, since control flow will return
+        # after the call)
+        return (
+            PpcArch.is_conditional_return_instruction(instr)
+            or PpcArch.is_branch_instruction(instr)
+            or instr.mnemonic in ("blr", "bctr")
+        )
+
+    @staticmethod
+    def is_delay_slot_instruction(instr: Instruction) -> bool:
+        return False
+
+    @staticmethod
+    def is_return_instruction(instr: Instruction) -> bool:
+        return instr.mnemonic == "blr"
+
+    @staticmethod
+    def is_conditional_return_instruction(instr: Instruction) -> bool:
+        return instr.mnemonic in (
+            "beqlr",
+            "bgelr",
+            "bgtlr",
+            "blelr",
+            "bltlr",
+            "bnelr",
+            "bnglr",
+            "bnllr",
+            "bnslr",
+            "bsolr",
+        )
+
+    @staticmethod
+    def is_jumptable_instruction(instr: Instruction) -> bool:
+        return instr.mnemonic == "bctr"
+
+    @staticmethod
+    def missing_return() -> List[Instruction]:
+        return [Instruction("blr", [], InstructionMeta.missing())]
+
+    INSTRS_R0_AS_ZERO: ClassVar[Set[str]] = {
```

Comment:
I originally generated it by looking for `(rA|0)` in the processor manual; but that missed some instructions like `lfs`. I need to go through this more carefully still.

Reply:
Updated the list more carefully, by reading each instruction's page in the processor manual. 

I added all instructions, even ones that I don't think will be implemented, so that we don't have to worry about keeping this list up to date in the future. (I also had to add support for `$r0` as the first arg, even though I don't think we'll ever implement any of those instructions.)